injector/FindLocalImports.py

```
@@ -0,0 +1,71 @@
+    """
+    rootDir = os.path.dirname(sourceFile)
+    pathsFound = []
+    fileQueue = [sourceFile]
+
+    while len(fileQueue) > 0:
+        sourceFile = fileQueue.pop()
+        try:
+            with open(sourceFile, "r") as f:
+                source = f.read()
+        except IOError as e:
+            raise IOError(f"Failed to read source file {sourceFile}: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)